src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -168,6 +168,11 @@ public string TestCode
         /// </summary>
         public List<Func<Solution, ProjectId, Solution>> SolutionTransforms { get; } = new List<Func<Solution, ProjectId, Solution>>();
 
+        /// <summary>
+        /// Gets the syntax trees generated by the source generators during the test run.
+        /// </summary>
+        public ImmutableArray<SyntaxTree> GeneratedTrees { get; private set; } = ImmutableArray<SyntaxTree>.Empty;
```

Comment:
📝 I have quite a few concerns about this. However, since no changes are required to implement the original feature, it might be easiest to just close this PR.

Reply:
See my comments in #1074. It should not be necessary to rewrite half of the test framework again and again, just to get hold of this simple thing.

Reply:
BTW, what are your concerns here?